### DISCUSSION TITLE
Equivalent of contentReplaced & willReplaceContent events in swup 4

### DISCUSSION THREAD

**Discussion by andresclua** [swup/swup#755] [Help] [answered]
hey there, switching to v4!
but ended up a little bit confusing, what are the hooks that are equivalent to this in v3?

```js
document.addEventListener("swup:contentReplaced", () => {
  this.contentReplaced();
});
document.addEventListener("swup:willReplaceContent", () => {
  this.willReplaceContent();
});
```
  
  thanks in advance!

**Comment by daun** [comment, marked as answer]
The equivalent is hooking before as well as after the `content:replace` event:

```js
// = willReplaceContent
swup.hooks.before('content:replace', () => {}) 
// = contentReplaced
swup.hooks.on('content:replace', () => {}) 
```

If you require DOM events, there currently is no way of hooking before an internal hook. That use case wasn't on our map so far.

**Reply by andresclua** [reply]
ended up with this!
```
// = contentReplaced
            this.swup.hooks.on('content:replace', () => {
                console.log('content replaced')
                this.contentReplaced() 
            }) 

            // = willReplaceContent
            this.swup.hooks.before('content:replace', () => {
                console.log('willReplaceContent')
                this.willReplaceContent()
            }) 
            
            ```

**Comment by hirasso** [comment]
@andresclua, congrats for making the switch to swup4 👏 

...would `visit:start` work for you as a replacement for the deprecated `willReplaceContent`? It's being fired every time a swup visit starts, before `content:replace`:

```js
document.addEventListener("swup:visit:start", () => {
  this.willReplaceContent();
});
```

Other candidates are `page:load`, `animation:out:start`, `animation:out:await` and `animation:out:end`. All these are being fired before `content:replace`.
